sdk/monitor/azure-monitor-query/src/samples/java/com/azure/monitor/query/LargeQuerySample.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.monitor.query;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.util.Configuration;
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.monitor.query.models.*;
+
+public class LargeQuerySample {
+
+
+    // BEGIN: com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+    static final String WORKSPACE_ID = "{workspace-id}";
+    // END: com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+
+
+    static LogsQueryClient client;
+
+    public static void main(String[] args) {
```

Comment:
Main method should demonstrate starting with a query that gets split into multiple sub-queries and show how to consolidate the results of sub-queries into a single result of the original query.